gui/window.go

```diff
@@ -603,6 +605,79 @@ func (window *Window) launchWelcomeView() {
 // launchMenuView launches the menu view
 func (window *Window) launchMenuView() {
 	window.menu.currentPage.StoreChanges()
+
+	// If syscheck fails, launch a dialog box and force the user to exit
+	if retErr := syscheck.RunSystemCheck(true); retErr != nil {
+		contentBox, err := gtk.BoxNew(gtk.ORIENTATION_HORIZONTAL, 10)
+		contentBox.SetHAlign(gtk.ALIGN_FILL)
+		contentBox.SetMarginBottom(common.TopBottomMargin)
+		if err != nil {
+			log.Warning("Error creating box")
+			return
+		}
+
+		//Custom CSS class for this dialog - make our error icon red(ish)
+		provider, err := gtk.CssProviderNew()
+		if err != nil {
+			log.Warning("Error creating CSS provider")
+			return
+		}
+		customTheme := ".dialog image {color: #FDB814;}"
+		if err := provider.LoadFromData(customTheme); err != nil {
+			log.Warning("Error creating CSS from internal data")
+			return
+		}
+
+		screen, err := gdk.ScreenGetDefault()
+		if err != nil {
+			log.Warning("Error getting screen")
+			return
+		}
+		gtk.AddProviderForScreen(screen, provider, gtk.STYLE_PROVIDER_PRIORITY_APPLICATION)
+
+		st, err := contentBox.GetStyleContext()
+		if err != nil {
+			log.Warning("Error getting style context")
+			return
+		}
+
+		// Style the dialog
+		st.AddClass("dialog")
+
+		icon, err := gtk.ImageNewFromIconName("dialog-error-symbolic", gtk.ICON_SIZE_DIALOG)
+		if err != nil {
+			log.Warning("gtk.ImageNewFromIconName failed for icon dialog-error-symbolic")
+			return
+		}
+
+		icon.SetMarginEnd(12)
+		icon.SetHAlign(gtk.ALIGN_START)
+		icon.SetVAlign(gtk.ALIGN_START)
+		contentBox.PackStart(icon, false, true, 0)
+
+		label, err := gtk.LabelNew("System failed to pass pre-install checks.\n\n" + retErr.Error())
+		if err != nil {
+			log.Warning("Error creating label")
+			return
+		}
+		label.SetUseMarkup(true)
+		label.SetHAlign(gtk.ALIGN_END)
+		contentBox.PackStart(label, false, true, 0)
+
+		dialog, err := common.CreateDialogOneButton(contentBox, "System Check Failed", utils.Locale.Get("EXIT"), "button-cancel")
```

Comment:
Please add entries in the en_US, es_MX and zh_CN locale files for all strings that is displayed in the installer.